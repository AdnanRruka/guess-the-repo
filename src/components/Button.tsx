import React from 'react';
import styled from 'styled-components/native';

const Wrapper = styled.TouchableOpacity`
  border-radius: 20px;
  background-color: #000000;
  padding: 10px 20px;
`;

const Label = styled.Text`
  font-size: 24px;
  color: #fff;
`;

interface ButtonProps {
  label: string;
  onPress: () => void;
}

const Button = (props: ButtonProps) => {
  return (
    <Wrapper onPress={props.onPress}>
      <Label>{props.label}</Label>
    </Wrapper>
  );
};

export default Button;
