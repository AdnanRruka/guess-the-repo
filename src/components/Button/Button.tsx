import React from 'react';
import styled from 'styled-components/native';

interface WrapperProps {
  marginBottom?: boolean;
}

const Wrapper = styled.TouchableOpacity`
  border-radius: 20px;
  background-color: #000000;
  padding: 10px 20px;
  ${(p: WrapperProps) => p.marginBottom && 'margin-bottom: 20px'}
`;

const Label = styled.Text`
  font-size: 24px;
  color: #fff;
`;

interface ButtonProps {
  label: string;
  marginBottom?: boolean;
  onPress: () => void;
}

export const Button = (props: ButtonProps) => {
  return (
    <Wrapper onPress={props.onPress} marginBottom={props.marginBottom}>
      <Label>{props.label}</Label>
    </Wrapper>
  );
};
