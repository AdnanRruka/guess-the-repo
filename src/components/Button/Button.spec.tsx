// Hello.spec.js
import React from 'react';
import {render, shallow} from 'react-native-testing-library';
import {Button} from './Button';

describe('Button', () => {
  it('renders correct', () => {
    const {queryByText} = render(<Button label="Button" onPress={() => {}} />);
    expect(queryByText('Button')).not.toBeNull();
  });
  it('should match to snapshot', () => {
    const component = shallow(<Button label="Button" onPress={() => {}} />);
    expect(component).toMatchSnapshot();
  });
});
