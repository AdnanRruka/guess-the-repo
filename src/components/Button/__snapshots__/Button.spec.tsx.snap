// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should match to snapshot 1`] = `
Object {
  "output": <Styled(TouchableOpacity)
    onPress={[Function]}
  >
    <Styled(Text)>
      Button
    </Styled(Text)>
  </Styled(TouchableOpacity)>,
}
`;
